### PR TITLE
Url encoding and display avatar in blog post

### PR DESCRIPTION
**Pull request by huytd**
Test Build: https://webuild-blog-aj8ebnpkh-kingbazoka.vercel.app

What changed:

- Added a Github action to run test on every pull requests
- A new scheme for the blog post URL, previously we put the full URL of the blog into the query param, with this change, we encode it with some other information such as author information, and use a base64 string for the encoded data, so it probably looks better (IMHO).
- Display the author's information in the post page

### CODE REVIEW

**Comment by huytd** [comment]
#43 will be fixed with this PR

**Comment by huytd** [comment]
PR finalized. You can start review now @lqt93

File: components/Pagination/index.tsx [hunk 1]
```diff
@@ -38,14 +45,13 @@ const ButtonArrow = ({ disabled, direction, onClick }) => {
     />
   );
 };
-
-const ArrowContainer = ({ available, direction, onClick }) => {
+const ArrowContainer = ({ disabled, direction, onClick }: ButtonArrowProps) => {
```
Comment (thien-do) [inline components/Pagination/index.tsx:48]: You guys didn't have strict mode? :|

Comment (huytd) [reply components/Pagination/index.tsx:48]: @nhducit added it just yesterday

File: utils/url.ts [hunk 2]
```diff
@@ -1,4 +1,27 @@
+import base64 from 'base-64';
```
Comment (thien-do) [inline utils/url.ts:1]: You may not need it https://stackoverflow.com/questions/30106476/using-javascripts-atob-to-decode-base64-doesnt-properly-decode-utf-8-strings

**Review by thien-do — APPROVED** [review]

**Comment by huytd** [comment]
The wait is no more

**Pull request merged**